e2e/imgbuild/imgbuild.go

```diff
@@ -1530,43 +1530,130 @@ echo 'export LEGACY_TEST_ENV=legacy-value' >> $SINGULARITY_ENVIRONMENT
 	)
 }
 
+func (c *imgBuildTests) testContainerBuildUnderFakerootModes(t *testing.T) {
+	tmpDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "test-container-build-under-fakeroot-modes-", "")
+	defer cleanup(t)
+
+	// running under the mode 1, 1a (--with-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--fakeroot", fmt.Sprintf("%s/openssh-mode1a.sif", tmpDir), "testdata/regressions/issue_476.def"),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 1, 1b (--without-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--fakeroot", fmt.Sprintf("%s/openssh-mode1b.sif", tmpDir), "testdata/regressions/issue_476.def"),
+		e2e.WithEnv([]string{"_APPTAINER_E2E_NOT_USE_SUID=1"}),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 2(https://apptainer.org/docs/user/main/fakeroot.html)
+	// temporally remove executable permission from fakeroot
+	e2e.Privileged(func(t *testing.T) {
+		cmd := exec.Command("chmod", "-x", "/usr/bin/fakeroot")
+		err := cmd.Run()
+		if err != nil {
+			t.Fatal(err)
+		}
+	})(t)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.FakerootModeTwoProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", fmt.Sprintf("%s/openssh-mode2.sif", tmpDir), "testdata/regressions/issue_476.def"),
+		e2e.WithEnv([]string{"_APPTAINER_E2E_FAKEROOT=1"}),
+		e2e.ExpectExit(255), // because chown will fail
+	)
+	// restore permission to fakeroot
+	e2e.Privileged(func(t *testing.T) {
+		cmd := exec.Command("chmod", "+x", "/usr/bin/fakeroot")
+		err := cmd.Run()
+		if err != nil {
+			t.Fatal(err)
+		}
+	})(t)
+
+	// running under the mode 3(https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.FakerootProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", fmt.Sprintf("%s/openssh-mode3.sif", tmpDir), "testdata/regressions/issue_476.def"),
+		e2e.WithEnv([]string{"_APPTAINER_E2E_FAKEROOT=1"}),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 4(https://apptainer.org/docs/user/main/fakeroot.html)
+	// disable the userns first
+	e2e.Privileged(func(t *testing.T) {
+		cmd := exec.Command("sysctl", "-w", "kernel.unprivileged_userns_clone=0")
```

Comment:
Doing this will affect tests running under `UserNamespaceProfile` that are running in parallel to those tests, we are going to get random failure

Reply:
thanks for the comment! I haven't thought about it.